src/bewit.rs

```diff
@@ -32,6 +32,47 @@ impl<'a> Bewit<'a> {
         }
     }
 
+    /// Extract the `bewit` query parameter, if any, from the path.  If the path contains no bewit,
+    /// the return value is `Ok(None)` and the path is not modified. If the path contains a valid
+    /// bewit, the path is modified and `Ok(Some(bewit))` returned.  If the path contains an
+    /// invalid bewit, the Result is an Err.
+    ///
+    /// The path manipulation is tested to correspond to that preformed by the hueniverse/hawk
+    /// implementation-specification)
+    pub fn from_path(path: &mut Cow<'a, str>) -> Result<Option<Bewit<'a>>> {
+        const PREFIX: &str = "bewit=";
+
+        let mut bewit_components: Vec<&str> = vec![];
+        let components: Vec<&str> = path
+            .split(|c| c == '&' || c == '?')
```

Comment:
IIUC, there's a risk that this will mis-parse paths that happen to use `&` in the resource component, such as:
```
https://example.com/this&that&bewit=true
```
Such paths seem vanishingly unlikely in practice, but I don't think anything outright prevents them, and I can imagine them being constructed maliciously to try to trick the application's validation logic. It may be worth splitting first on the `?` and then on the `&`, just in case.

Reply:
Or using `url` which is already a dependency.

Reply:
This is a good point.  I filed #25 for this.